code/game/atoms.dm

```diff
@@ -48,6 +48,8 @@ var/global/list/ghdel_profiling = list()
 	//var/harm_label_icon_state //Makes sense to have this, but I can't sprite. May be added later.
 	var/list/last_beamchecks // timings for beam checks.
 
+	var/vending_cat = null// subcategory for vending machines.

```

Comment:
This is really not how I would implement this.
Ideally, products should be a list of new vending datums which you can pass an arg like category and price to.